lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -95,51 +95,51 @@ func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_pro
 	return listRes, nil
 }
 
-func (s *subscriberdbServicer) getDigestAndNoUpdates(
+// getDigestInfo returns the correctly formatted SubscribersDigest and NoUpdates
+// values in response to a list subscribers request.
+func (s *subscriberdbServicer) getDigestInfo(
 	req *lte_protos.ListSubscribersRequest,
 	networkID string,
 	apnsByName map[string]*lte_models.ApnConfiguration,
 	apnResourcesByAPN lte_models.ApnResources,
 ) (*lte_protos.SubscribersDigest, bool) {
-
-	digest, noUpdates := req.Digest, false
-	// This functionality is currently placed behind a feature flag.
-	if s.flatDigestEnabled && req.PageToken == "" {
-		digestString, err := getDigest(networkID, apnsByName, apnResourcesByAPN)
-		// If digest generation fails, the error is swallowed to not affect the main functionality.
-		if err != nil {
-			glog.Errorf("Generating digest for subscribers in network of gateway %s failed", networkID)
-		} else {
-			digest = &lte_protos.SubscribersDigest{Md5Base64Digest: digestString}
-			noUpdates = req.Digest.Md5Base64Digest == digestString
-		}
+	digestProto, noUpdates := req.Digest, false
+	// This functionality is currently placed behind a feature flag
+	if !s.flatDigestEnabled || req.PageToken != "" {
+		return digestProto, noUpdates
 	}
 
-	return digest, noUpdates
+	digest, err := getDigest(networkID, apnsByName, apnResourcesByAPN)
+	// If digest generation fails, the error is swallowed to not affect the main functionality
+	if err != nil {
+		glog.Errorf("Generating digest for subscribers in network of gateway %s failed", networkID)
+	} else {
+		digestProto = &lte_protos.SubscribersDigest{Md5Base64Digest: digest}
+		noUpdates = req.Digest.Md5Base64Digest == digest
+	}
+	return digestProto, noUpdates
 }
 
-// getDigest loads all subscribers registered on the current network, and returns
-// a deterministic subscriber flat digest.
+// getDigest returns a deterministic digest of all subscribers in the network.
+// Digest is generated by loading all subscribers in the network.
 func getDigest(
 	networkID string,
 	apnsByName map[string]*lte_models.ApnConfiguration,
 	apnResourcesByAPN lte_models.ApnResources,
 ) (string, error) {
 	subProtosById := map[string]proto.Message{}
 	curToken := ""
-	for {
-		subProtos, nextToken, err := loadSubProtosPage(0, curToken, networkID, apnsByName, apnResourcesByAPN)
+	nextToken := "init"
+	for nextToken != "" {
+		subProtos, token, err := loadSubProtosPage(0, curToken, networkID, apnsByName, apnResourcesByAPN)
 		if err != nil {
 			return "", err
 		}
 		for _, subProto := range subProtos {
-			index := subProto.Sid.Id
-			subProtosById[index] = subProto
-		}
-		if nextToken == "" {
-			break
+			subProtosById[subProto.Sid.Id] = subProto
 		}
-		curToken = nextToken
+		curToken = token
+		nextToken = token
```

Comment:
I don't know about the "nextToken" pattern, using "init" is a little unclear

Try something like

```
foundEmptyToken := false
for !foundEmptyToken {
	// ...
	foundEmptyToken = nextToken == "" 
}
```